backend/stakepoold/server.go

```diff
@@ -338,27 +337,10 @@ func runMain() error {
 		}
 	}
 
-	// Only accept a single CTRL+C
-	c := make(chan os.Signal, 1)
-	signal.Notify(c, os.Interrupt)
-
-	// Start waiting for the interrupt signal
-	go func() {
-		<-c
-		signal.Stop(c)
-		// Close the channel so multiple goroutines can get the message
-		log.Info("CTRL+C hit.  Closing goroutines.")
-		// Stop autoreconnect.
-		ctx.WalletConnection.Stop()
-
-		saveData(ctx)
-		close(ctx.Quit)
-	}()
-
 	ctx.Wg.Add(3)
-	go ctx.NewTicketHandler()
-	go ctx.SpentmissedTicketHandler()
-	go ctx.WinningTicketHandler()
+	go ctx.NewTicketHandler(shutdownContext.Done())
+	go ctx.SpentmissedTicketHandler(shutdownContext.Done())
+	go ctx.WinningTicketHandler(shutdownContext.Done())
```

Comment:
Should be possible for each of these methods to create their own shutdownContext using `withShutdownCancel`, otherwise this isn't much different from what we currently have.

Reply:
It is the same. This is about readability and standardization.

Reply:
Oh, I see. We could go the extra mile and have those 3 methods and any other method that requires a context, create their own shutdownContext using `withShutdownCancel`. What do you think?

Reply:
hmmm, idk. Would that be better? I'll look a little more into when the other decred repo's might spawn a new context. When I say standardization, I mostly mean what wallet and dcrd are doing. I think there a lot of good ideas that have developed that we could make use of.  Although this particular refactor has been around since btcd apparently, just without the context. It was just a channel at first. 
https://github.com/decred/dcrwallet/commit/b145868a4bc5e8bd9fec82afc0a806e0d2529f67#diff-dca9e25a1ecb5a565b1bf9142c755f71